Extract validator logic into a trait.

markstory opened the pull request:
Having these methods in a trait will let us not copy + paste code into the elastic search plugin. It could also be useful for other datasource plugins that need validation features.

markstory committed:
Extract validator logic into a trait.

Having these methods in a trait will let us not copy + paste code into
the elastic search plugin. It could also be useful for other datasource
plugins that need validation features.

lorenzo commented on src/Validation/ValidatorAwareTrait.php:
```diff
@@ -0,0 +1,108 @@
+            }
+        }
+
+        $validator->provider('table', $this);
```
hmm, this should probably be done somewhere else.

markstory replied:
I think the name is the only problem. What if implementing classes had to define a property for the provider name? Implementers also need to provide a constant which is unfortunate.

lorenzo replied:
Adding the property to the class could work fine

markstory replied:
Ok, I'll do that.

markstory replied:
Updated :smile:

markstory committed:
Add docs and constant for defining validator hookups.

Document how the trait interacts with the including class, and define
a constant so that not all validator aware things get assigned as
'table' to the validators.

lorenzo commented:
Good job!
